torchinductor/codegen/triton.py

```diff
@@ -162,6 +162,14 @@ def randn(seed, offset, _):  # _ here to keep the contract identical to CPU rand
     def rsqrt(x):
         return f"tl.libdevice.rsqrt({x})"
 
+    @staticmethod
+    def signbit(x):
+        return f"{x} < 0"
```

Comment:
isn't there a `libdevice.signbit`? Oh ok there is but separate for floating and double types, we can keep this.

Reply:
Perhaps
```suggestion
        return f""tl.libdevice.signbitd({x}) if {x}.dtype is tl.float64 else tl.libdevice.signbitf({x})"
```

Reply:
Where is there  alist ofr all the functions in `libdevice`? Do I have to `dir()` it or what's the best way to inspect it?
Also, I'll use Jansel's proposed pattern + a `x < 0` fallback for signed integers.

Reply:
I look at [this file](https://github.com/openai/triton/blob/master/python/triton/language/libdevice.py) to see the list of functions.
```
tl.libdevice.signbitf({x}) if {x}.dtype is tl.float32 else {x} < 0
```
Might be simpler and support ints too.  float32 is the main thing we care about inside a kernel.

Reply:
That one would be wrong for floating points of size other than 32 and the value `-0.0` but I guess that's alright. Otherwise, if we wanted to go with the definition from cppref, we should probably do
```python
return f"tl.libdevice.signbitf({x}) if {x}.dtype is tl.float32 else tl.libdevice.signbitd(to_dtype({x}, torch.float64))"
```

Reply:
float16/bfloat32 are up cast to float32 inside the kernel, so the only cases are float32, float64, and the integers.